### PR TITLE
feat(open-ai-official): add reasoning summaries to Responses streaming

### PR DESCRIPTION
**Pull request by aavetis**
## Issue
Related to #3252, #3637, and the closed PR #4489 (this PR follows the official OpenAI SDK path instead of Azure OpenAI SDK).

## Change
- Add reasoning summary support to the OpenAI Official Responses streaming model.
- Surface summary content in `AiMessage.thinking`.
- Add integration tests for reasoning summaries (positive and negative cases, gated by `OPENAI_API_KEY`).
- Document reasoning summaries in the OpenAI Official SDK docs.

## General checklist
- [X] There are no breaking changes (API, behaviour)
- [X] I have added unit and/or integration tests for my change
- [X] The tests cover both positive and negative cases
- [X] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [X] I have manually run all the unit and integration tests in the core and main modules, and they are all green
- [X] I have added/updated the documentation
- [ ] I have added an example in the examples repo (only for "big" features)
- [ ] I have added/updated Spring Boot starter(s) (if applicable)

## Checklist for adding new maven module
- [ ] I have added my new module in the root `pom.xml` and `langchain4j-bom/pom.xml`

## Checklist for adding new embedding store integration
- [ ] I have added a `{NameOfIntegration}EmbeddingStoreIT` that extends from either `EmbeddingStoreIT` or `EmbeddingStoreWithFilteringIT`
- [ ] I have added a `{NameOfIntegration}EmbeddingStoreRemovalIT` that extends from `EmbeddingStoreWithRemovalIT`

## Checklist for changing existing embedding store integration
- [ ] I have manually verified that the `{NameOfIntegration}EmbeddingStore` works correctly with the data persisted using the latest released version of LangChain4j

### CODE REVIEW

File: langchain4j-open-ai-official/src/main/java/dev/langchain4j/model/openaiofficial/OpenAiOfficialResponsesStreamingChatModel.java [hunk 1]
```diff
@@ -114,6 +115,7 @@ public class OpenAiOfficialResponsesStreamingChatModel implements StreamingChatM
     private final String promptCacheKey;
     private final String promptCacheRetention;
     private final String reasoningEffort;
```
Comment (jdubois) [inline langchain4j-open-ai-official/src/main/java/dev/langchain4j/model/openaiofficial/OpenAiOfficialResponsesStreamingChatModel.java]: Can we uses com.openai.models.ReasoningEffort instead of a String? This is what we do in OpenAiOfficialImageModel for the size and quality.

Comment (jdubois) [reply langchain4j-open-ai-official/src/main/java/dev/langchain4j/model/openaiofficial/OpenAiOfficialResponsesStreamingChatModel.java]: Same as above, can we use a com.openai.models.Reasoning.Summary instead of a String?

File: langchain4j-open-ai-official/src/main/java/dev/langchain4j/model/openaiofficial/OpenAiOfficialResponsesStreamingChatModel.java [hunk 2]
```diff
@@ -983,6 +1004,50 @@ private void extractTokenUsageAndComplete(com.openai.models.responses.Response r
...
+
+        private static String extractReasoningSummary(com.openai.models.responses.Response response) {
+            if (response == null) {
+                return null;
+            }
+            List<ResponseOutputItem> outputItems;
+            try {
+                outputItems = response.output();
+            } catch (RuntimeException ex) {
+                return null;
```
Comment (jdubois) [inline langchain4j-open-ai-official/src/main/java/dev/langchain4j/model/openaiofficial/OpenAiOfficialResponsesStreamingChatModel.java:1071]: I don't like to ignore exceptions, can we log it at least?

File: langchain4j-open-ai-official/src/main/java/dev/langchain4j/model/openaiofficial/OpenAiOfficialResponsesStreamingChatModel.java [hunk 3]
```diff
@@ -983,6 +1004,50 @@ private void extractTokenUsageAndComplete(com.openai.models.responses.Response r
...
+                        if (text == null || text.isBlank()) {
+                            continue;
+                        }
+                        if (!builder.isEmpty()) {
+                            builder.append("\n");
+                        }
+                        builder.append(text);
+                    }
+                } catch (RuntimeException ignored) {
+                    // Summary is optional and may be missing or malformed in some responses.
```
Comment (jdubois) [inline langchain4j-open-ai-official/src/main/java/dev/langchain4j/model/openaiofficial/OpenAiOfficialResponsesStreamingChatModel.java]: Same as above, can we at least log it instead of ignoring the error totally?

File: langchain4j-open-ai-official/src/main/java/dev/langchain4j/model/openaiofficial/OpenAiOfficialResponsesStreamingChatModel.java [hunk 4]
```diff
@@ -983,6 +1004,50 @@ private void extractTokenUsageAndComplete(com.openai.models.responses.Response r
...
+            List<ResponseOutputItem> outputItems;
+            try {
+                outputItems = response.output();
+            } catch (RuntimeException ex) {
+                return null;
+            }
+            if (outputItems == null || outputItems.isEmpty()) {
+                return null;
+            }
+            StringBuilder builder = new StringBuilder();
```
Comment (jdubois) [inline langchain4j-open-ai-official/src/main/java/dev/langchain4j/model/openaiofficial/OpenAiOfficialResponsesStreamingChatModel.java]: "builder" is too generic here, can you rename it to something like `responseStringBuilder`? Also with modern Java, you don't gain from using a StringBuilder, so maybe just a String would be easier.

File: langchain4j-open-ai-official/src/test/java/dev/langchain4j/model/openaiofficial/openai/OpenAiOfficialResponsesStreamingChatModelIT.java [hunk 5]
```diff
@@ -367,6 +367,53 @@ void should_support_reasoning_effort() {
         assertThat(handler.get().aiMessage().text()).contains("Paris");
     }
 
+    @Test
+    void should_support_reasoning_summary() {
+
+        // given
+        var client = OpenAIOkHttpClient.builder()
```
Comment (jdubois) [inline langchain4j-open-ai-official/src/test/java/dev/langchain4j/model/openaiofficial/openai/OpenAiOfficialResponsesStreamingChatModelIT.java]: Can you use the same system as in the other tests? Like in `OpenAiOfficialChatModelIT`? This will ensure it works the same everywhere, including on Microsoft Foundry

File: langchain4j-open-ai-official/src/test/java/dev/langchain4j/model/openaiofficial/openai/OpenAiOfficialResponsesStreamingChatModelIT.java [hunk 6]
```diff
@@ -367,6 +367,53 @@ void should_support_reasoning_effort() {
...
+                .apiKey(System.getenv("OPENAI_API_KEY"))
```
Comment (jdubois) [inline langchain4j-open-ai-official/src/test/java/dev/langchain4j/model/openaiofficial/openai/OpenAiOfficialResponsesStreamingChatModelIT.java]: Also this is copied everywhere in the test, can you have a look at how I do it in `InternalOpenAiOfficialTestHelper`? This will be more consistent, and remove this duplication

File: langchain4j-open-ai-official/src/test/java/dev/langchain4j/model/openaiofficial/openai/OpenAiOfficialResponsesStreamingChatModelIT.java [hunk 7]
```diff
@@ -443,6 +490,5 @@ protected boolean supportsPartialToolStreaming(dev.langchain4j.model.chat.Stream
 
     @Override
     @Disabled("Can't do it reliably")
-    protected void should_execute_multiple_tools_in_parallel_then_answer(StreamingChatModel model) {
-    }
+    protected void should_execute_multiple_tools_in_parallel_then_answer(StreamingChatModel model) {}
```
Comment (jdubois) [inline langchain4j-open-ai-official/src/test/java/dev/langchain4j/model/openaiofficial/openai/OpenAiOfficialResponsesStreamingChatModelIT.java]: I've run it a few times without any issue, is this when you run it alone, or with other tests?

**Review by jdubois — CHANGES_REQUESTED** [review]
Most of my comments are to have something which is more coherent with the other tests. Also, it should work automatically on Microsoft Foundry if you follow what I did - in the end, OpenAI also works on the same platform, so it's just a matter of having the right URL/API key.
I did ran all the tests of Microsoft Foundry, and I only had 3 errors:
- `should_respect_modelName_in_default_model_parameters` which fails with a 404: I always struggle with that one on Microsoft Foundry, so I'm not sure about this one.
- `should_accept_multiple_images_as_public_URLs` and `should_accept_single_image_as_public_URL` but I'm pretty sure it's an issue on my end, we don't allow the public URLs anymore (not the first time I see this, so I'm not worried)

File: langchain4j-open-ai-official/src/main/java/dev/langchain4j/model/openaiofficial/OpenAiOfficialResponsesStreamingChatModel.java (hunk 1 above)
Comment (aavetis) [reply langchain4j-open-ai-official/src/main/java/dev/langchain4j/model/openaiofficial/OpenAiOfficialResponsesStreamingChatModel.java]: `reasoningEffort` is now stored as `com.openai.models.ReasoningEffort`. I kept a small convenience overload `reasoningEffort(String)` similar to `OpenAiOfficialImageModel`’s typed builder pattern.

`reasoningSummary` is now stored as `com.openai.models.Reasoning.Summary`. Kept `reasoningSummary(String)` as a convenience wrapper for consistency with other typed builders.

Happy to remove the wrapper/overload if we want only the typed setter.

File: langchain4j-open-ai-official/src/main/java/dev/langchain4j/model/openaiofficial/OpenAiOfficialResponsesStreamingChatModel.java (hunk 2 above)
Comment (aavetis) [reply langchain4j-open-ai-official/src/main/java/dev/langchain4j/model/openaiofficial/OpenAiOfficialResponsesStreamingChatModel.java:1071]: We now log at debug if reading `response.output()` fails during reasoning summary extraction.

File: langchain4j-open-ai-official/src/main/java/dev/langchain4j/model/openaiofficial/OpenAiOfficialResponsesStreamingChatModel.java (hunk 3 above)
Comment (aavetis) [reply langchain4j-open-ai-official/src/main/java/dev/langchain4j/model/openaiofficial/OpenAiOfficialResponsesStreamingChatModel.java]: We now log at debug the per-output-item extraction failures instead of ignoring them completely. Summary remains optional, so we still degrade gracefully to null.

File: langchain4j-open-ai-official/src/main/java/dev/langchain4j/model/openaiofficial/OpenAiOfficialResponsesStreamingChatModel.java (hunk 4 above)
Comment (aavetis) [reply langchain4j-open-ai-official/src/main/java/dev/langchain4j/model/openaiofficial/OpenAiOfficialResponsesStreamingChatModel.java]: Replaced the generic builder with a clearer approach to improve readability and avoid manual `StringBuilder` concatenation.

File: langchain4j-open-ai-official/src/test/java/dev/langchain4j/model/openaiofficial/openai/OpenAiOfficialResponsesStreamingChatModelIT.java (hunk 5 above)
Comment (aavetis) [reply langchain4j-open-ai-official/src/test/java/dev/langchain4j/model/openaiofficial/openai/OpenAiOfficialResponsesStreamingChatModelIT.java]: Refactored the Responses-streaming ITs to follow the same helper-driven setup pattern as the other OpenAI official ITs. I also validated locally on Microsoft Foundry. Using the endpoint and appending /openai/v1 worked for the Responses API (e.g. https://<resource>.services.ai.azure.com/openai/v1).

File: langchain4j-open-ai-official/src/test/java/dev/langchain4j/model/openaiofficial/openai/OpenAiOfficialResponsesStreamingChatModelIT.java (hunk 6 above)
Comment (aavetis) [reply langchain4j-open-ai-official/src/test/java/dev/langchain4j/model/openaiofficial/openai/OpenAiOfficialResponsesStreamingChatModelIT.java]: Moved repeated client/model creation into `InternalOpenAiOfficialTestHelper` and updated the tests to reuse it.

File: langchain4j-open-ai-official/src/test/java/dev/langchain4j/model/openaiofficial/openai/OpenAiOfficialResponsesStreamingChatModelIT.java (hunk 7 above)
Comment (aavetis) [reply langchain4j-open-ai-official/src/test/java/dev/langchain4j/model/openaiofficial/openai/OpenAiOfficialResponsesStreamingChatModelIT.java]: The test assumes parallel tool calls come back in a deterministic order. Instead of relying on the test’s fixed ordering (or keeping this scenario disabled), I updated `OpenAiOfficialResponsesStreamingChatModelIT.should_execute_multiple_tools_in_parallel_then_answer(...)` to keep the scenario covered but make assertions order-insensitive. We assert both `getWeather` and `getTime` tool calls are present (by name + args) and then verify the follow-up answer after providing both tool results, without relying on call order/index. This should remove intermittent failures when running the full suite.

**Comment by aavetis** [comment]
Thanks for the comments. Happy to adjust as needed

File: langchain4j-open-ai-official/src/main/java/dev/langchain4j/model/openaiofficial/OpenAiOfficialResponsesStreamingChatModel.java (hunk 4 above)
Comment (tkumar1918) [reply langchain4j-open-ai-official/src/main/java/dev/langchain4j/model/openaiofficial/OpenAiOfficialResponsesStreamingChatModel.java]: > "builder" is too generic here, can you rename it to something like `responseStringBuilder`? Also with modern Java, you don't gain from using a StringBuilder, so maybe just a String would be easier.

Since this code aggregates multiple streamed parts inside nested loops, using StringBuilder avoids creating intermediate String instances on each append. Modern Java optimizes single-expression concatenation, but not repeated += in loops. So StringBuilder is still appropriate here.

**Comment by mikehearn** [comment]
I tried this out and it worked, although I found I preferred a slightly different design in which `onPartialThinking` is invoked in `handleReasoningSummaryTextDone`. In practice, OpenAI models emit quite short thinking strings so streaming them is of no great value, and it wasn't obvious to me how to know when the thinking tokens are completed otherwise. By moving where the handler is invoked you get the full string which is normally (on 5.3-codex) a short piece of Markdown, and so there's no need to accumulate tokens yourself.

**Comment by mikehearn** [comment]
Actually I've now seen the model produce longer thinking strings. But with the current API there's no way to know when the model has finished emitting reasoning commentary, because there's no `onThinking` callback. So ideally we'd have both but if not then it might also be OK to just pass whole strings through to onPartialThinking. The current design is hard to understand though because when do you stop digesting the whole thought - the model is allowed to emit several different thoughts before it starts sending the final answer.

**Comment by aavetis** [comment]
i'm happy to go whichever directions the team wants. given it's close to the milestone end, need some guidance here

**Comment by jdubois** [comment]
@aavetis I will have time to work on this at the end of next week, and on the week after: I just realized by looking at your profile that you work at Microsoft ! Can you ping me on Teams? I couldn't find your real name, but on my side I should be easy to find... We can discuss this directly.

**Comment by dliubarskyi** [comment]
@aavetis thanks a lot! Last days I've been working on improving OpenAI Responses API support in both unofficial and official modules, so there are many conflicts, sorry about that. Do you mind if I'll close this PR in favour of https://github.com/langchain4j/langchain4j/pull/4954? In addition to reasoning summaries, it also adds support for encrypted reasoning and sync model

**Comment by dliubarskyi** [comment]
Closing in favour of https://github.com/langchain4j/langchain4j/pull/4954

@aavetis thank you!
